tensorflow/core/common_runtime/process_function_library_runtime.cc

```diff
@@ -637,6 +636,19 @@ Status ProcessFunctionLibraryRuntime::InstantiateMultiDevice(
   TF_RETURN_IF_ERROR(
       PartitionFunctionGraph(device_set, std::move(graph), &subgraphs));
 
+  for (const auto& pair : subgraphs) {
+    DumpGraph("Before running POST_PARTITIONING passes", pair.second.get());
+  }
+  optimization_options.graph = nullptr;
```

Comment:
Set device_set to nullptr as well. POST_PARTITIONING passes are not supposed to use devices.